### PR TITLE
Check deployment exists before getting operations. 

### PR DESCRIPTION
**Pull request by TianoMS**
Fix issue: https://github.com/Azure/azure-powershell/issues/2416. We need to make sure the deployment exists before trying to get any resources related to it.

### CODE REVIEW

**Comment by TianoMS** [comment]
On demand job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/892/

**Comment by vivsriaus** [comment]
@TianoMS What do you think about moving the checkExistence call to WriteDeploymentProgress() instead? This will ensure we always check for existence of deployment before listing operations. For instance, the current logic will throw the same error (deployment not found) in this instance:
1. Create deployment via powershell
2. Cancel running deployment via portal/CLI/another session/etc

**Comment by TianoMS** [comment]
@vivsriaus Thanks for pointing it out. I made this change because I saw we may also run into this error in "WaitDeploymentStatus" when we do "ResourceManagementClient.Deployments.Get(resourceGroup, deploymentName);"

Now it seems we need to check existence everywhere. Not sure if this is too much overhead.

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkClient/ResourceManagerSdkClient.cs [hunk 1]
```diff
@@ -312,23 +312,33 @@ public static List<string> ParseDetailErrorMessage(string statusMessage)
             Action<string, string, Deployment> job,
             params ProvisioningState[] status)
         {
-            DeploymentExtended deployment;
+            DeploymentExtended deployment = null;
+            bool deploymentExists = false;
             int counter = 5000;
 
-            do
+            while (deployment == null ||
```
Comment (gandhiniraj) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkClient/ResourceManagerSdkClient.cs]: deployment != null

Comment (TianoMS) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkClient/ResourceManagerSdkClient.cs]: Please ignore this change for now. I did some more investigation on this and I'm refactoring this change. Will update soon.

**Comment by TianoMS** [comment]
Updated a new iteration.

**Comment by vivsriaus** [comment]
@TianoMS LGTM.

**Comment by markcowl** [comment]
on demand run here:  http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/899/

**Comment by vivsriaus** [comment]
@TianoMS looks like we need to add the checkExistence before we GET a deployment as well. Let's discuss this before we merge the PR.

**Comment by markcowl** [comment]
On hold pending the discussion @vivsriaus suggests

**Comment by TianoMS** [comment]
On demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/900/

**Comment by TianoMS** [comment]
Will open another PR for 1.5.1.

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkClient/ResourceManagerSdkClient.cs [hunk 2]
```diff
@@ -351,21 +351,27 @@ private List<DeploymentOperation> GetNewOperations(List<DeploymentOperation> old
                     Enum.TryParse<HttpStatusCode>(operation.Properties.StatusCode, out statusCode);
                     if (!statusCode.IsClientFailureRequest())
                     {
-                        List<DeploymentOperation> newNestedOperations = new List<DeploymentOperation>();
+                        var resourceGroupName = ResourceIdUtility.GetResourceGroupName(operation.Properties.TargetResource.Id);
+                        var deploymentName = operation.Properties.TargetResource.ResourceName;
```
Comment (gandhiniraj) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkClient/ResourceManagerSdkClient.cs]: resourceName?

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkClient/ResourceManagerSdkClient.cs [hunk 3]
```diff
@@ -351,21 +351,27 @@ private List<DeploymentOperation> GetNewOperations(List<DeploymentOperation> old
...
+                                resourceGroupName: resourceGroupName,
+                                deploymentName: deploymentName);
 
-                        foreach (DeploymentOperation op in newNestedOperations)
-                        {
-                            DeploymentOperation nestedOperationWithSameIdAndProvisioningState = newOperations.Find(o => o.OperationId.Equals(op.OperationId) && o.Properties.ProvisioningState.Equals(op.Properties.ProvisioningState));
+                            newNestedOperations = GetNewOperations(operations, result);
 
-                            if (nestedOperationWithSameIdAndProvisioningState == null)
+                            foreach (DeploymentOperation op in newNestedOperations)
```
Comment (gandhiniraj) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkClient/ResourceManagerSdkClient.cs]: nit: deploymentOperation - just to follow the pattern

**Pull request closed without merging**